extensions/ql-vscode/test/vscode-tests/no-workspace/data-extensions-editor/model-details/model-details-panel.test.ts

```diff
@@ -0,0 +1,73 @@
+import { window, TreeView } from "vscode";
+import { CodeQLCliServer } from "../../../../../src/codeql-cli/cli";
+import { ExternalApiUsage } from "../../../../../src/data-extensions-editor/external-api-usage";
+import { ModelDetailsPanel } from "../../../../../src/data-extensions-editor/model-details/model-details-panel";
+import { DatabaseItem } from "../../../../../src/databases/local-databases";
+import { mockedObject } from "../../../utils/mocking.helpers";
+import {
+  createExternalApiUsage,
+  createUsage,
+} from "../../../../factories/data-extension/external-api-factories";
+
+describe("ModelDetailsPanel", () => {
+  const mockCliServer = mockedObject<CodeQLCliServer>({});
+  const dbItem = mockedObject<DatabaseItem>({
+    getSourceLocationPrefix: () => "test",
+  });
+
+  describe("setState", () => {
+    const hideModeledApis: boolean = false;
+    const externalApiUsages: ExternalApiUsage[] = [createExternalApiUsage()];
+
+    it("should update the tree view with the correct batch number", async () => {
+      const mockTreeView = {
+        badge: undefined,
+      } as TreeView<unknown>;
+      jest.spyOn(window, "createTreeView").mockReturnValue(mockTreeView);
+
+      const panel = new ModelDetailsPanel(mockCliServer);
+      await panel.setState(externalApiUsages, dbItem, hideModeledApis);
+
+      expect(mockTreeView.badge?.value).toBe(1);
+    });
+  });
+
+  describe("revealItem", () => {
+    let mockTreeView: TreeView<unknown>;
+
+    const hideModeledApis: boolean = false;
+    const usage = createUsage();
+
+    beforeEach(() => {
+      mockTreeView = {
+        reveal: jest.fn(),
+      } as unknown as TreeView<unknown>;
```

Comment:
This is similar to `as any as X` so we should ideally avoid it. Is there a way to get around it easily, or is creating mock `TreeView<>T>` items too painful? I'm happy to take it on as tech debt too.

Reply:
Lets leave it as is. I imagine creating the mock properly isn't trivial.

Reply:
Thanks for talking this through. I think `mockedObject` could be a good solution for this case. I've opened https://github.com/github/vscode-codeql/pull/2768 for this where I've tried to explain my reasoning, so let me know what you think.